README.md

```diff
@@ -1,133 +1,616 @@
+The Smithy TypeScript `typescript-client-codegen` code generator in this repository generates TypeScript clients from Smithy models, and can be built with both the idiomatic [Smithy CLI](#using-smithy-typescript-with-the-smithy-cli) or through [Gradle](#using-smithy-typescript-with-gradle).
+
+> The Smithy CLI is a prerequisite for this section when using the `smithy init` commands. See [the installation guide](https://smithy.io/2.0/guides/smithy-cli/cli_installation.html) for how to install the Smithy CLI. If installing the Smithy CLI is not preferred, the templates used can be found in the [Smithy Examples repository](https://github.com/smithy-lang/smithy-examples).
+
+For additional configuration, see [the `typescript-client-codegen` documentation](#client-sdk-code-generation-typescript-client-codegen-plugin) and [the documentation for `smithy-build.json`](https://smithy.io/2.0/guides/building-models/build-config.html).
+
+### Using Smithy TypeScript with the Smithy CLI
+
+Using the Smithy CLI, a new Smithy CLI project can be created using the default `smithy init` template. In this example, the project will be called `smithy-typescript-example-client`.
+
+```shell
+smithy init -o smithy-typescript-example-client
+cd smithy-typescript-example-client/
+```
+
+This will create a project with the following directory structure:
+
+```text
+smithy-typescript-example-client/
+├── README.md
+├── models
+│   └── weather.smithy
+└── smithy-build.json
+```
+
+To add a minimal `typescript-client-codegen` plugin, add the following to `smithy-build.json`: 
+
+```json
+// smithy-build.json
+{
+    "version": "1.0",
+    "sources": ["models"],
+    // Add the Smithy TypeScript code generator dependency
+    "maven": {
+        "dependencies": [
+            "software.amazon.smithy.typescript:smithy-typescript-codegen:0.19.0"
+        ]
+    },
+    "plugins": {
+        // Add the Smithy TypeScript client plugin
+        "typescript-client-codegen": {
+            // Minimal configuration: add package name and version
+            "package": {
+                "package": "@smithy/typescript-example-client",
+                "packageVersion": "0.0.1"
+            }
         }
-        dependencies {
-            "classpath"("software.amazon.smithy:smithy-cli:[1.33.0,1.34.0[")
+    }
+}
+```
+
+After `smithy-build.json` has been configured, run `smithy build`. This will code generate the TypeScript client under the `source` projection, found in the `build/smithy/source/typescript-client-codegen` directory.
+
+Verify the client is able to compile by running the following:
+
+```shell
+cd build/smithy/source/typescript-client-codegen
+# Yarn is used in this example, but equivalent commands using other package managers can be used, e.g. npm and pnpm
+yarn
+yarn build
+```
+
+> Note that running the NPM scripts to verify the generated TypeScript client is NOT part of the code generation process, and needs to be explicitly executed after the client is generated.
```

Comment:
What would the next step be? How would a customer take this output and start using it in an actual application like a react SPA?

Reply:
This section is just for generating a client as quickly as possible.

I think how to use packages is very customer-dependent and nuanced, and could be done in a follow-up PR.

Reply:
Let's revisit after we add the example to the examples repo